File: app2/src/main/java/com/kborid/setting/ui/MainActivity.java
package com.kborid.setting.ui;

import android.content.Intent;
import android.os.Bundle;

import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.MutableLiveData;

import com.kborid.setting.R;
import com.kborid.setting.databinding.ActivityMainBinding;
import com.kborid.setting.pojo.AddressInfo;
import com.kborid.setting.viewmodel.MainViewModel;
import com.kborid.setting.viewmodel.ViewModelsProvider;
import com.thunisoft.common.base.BaseSimpleActivity;
import com.thunisoft.common.tool.UIHandler;
import com.thunisoft.common.util.ToastUtils;

public class MainActivity extends BaseSimpleActivity {

    MainViewModel mainViewModel;
    ActivityMainBinding mBinding;

    @Override
    protected int getLayoutResId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initDataAndEvent(Bundle savedInstanceState) {
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        mBinding.setAddress(new AddressInfo("成都", "四川"));
        mBinding.setHandle(this);
        mainViewModel = ViewModelsProvider.of(this).get(MainViewModel.class);
        mBinding.setMainViewModel(mainViewModel);
        mBinding.setLifecycleOwner(this);
    }

    public void doClick(String msg) {
        mBinding.getAddress().setTest(new MutableLiveData<>("威海"));
        ToastUtils.showToast(msg + "我被点击了");
    }

    public void onBtnClick() {
        Intent intent1 = new Intent(this, TransActivity.class);
        startActivity(intent1);
        UIHandler.postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent2 = new Intent(MainActivity.this, FragmentActivity.class);
                startActivity(intent2);
            }
        }, 1000);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mBinding.unbind();
    }
}

File: kotlin/src/main/java/com/kborid/kotlin/ui/MainActivity.kt
package com.kborid.kotlin.ui

import android.os.Bundle
import com.kborid.kotlin.R
import com.thunisoft.common.base.BaseSimpleActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.toast

class MainActivity : BaseSimpleActivity() {

    val TAG: String = MainActivity::class.java.simpleName

    override fun getLayoutResId(): Int {
        return R.layout.activity_main
    }

    override fun initDataAndEvent(savedInstanceState: Bundle?) {
        tv_name.text = "世界，你好"
        runOnUiThread { toast(tv_name.text) }

        tv_name.setOnClickListener {
            tv_name.text = "了解世界"
            toast(tv_name.text)
        }


        var count = 0;
        tv_name.setOnLongClickListener {
            tv_name.text = when (count) {
                1, 3, 5, 7, 9 -> "凉风有信"
                in 13..19 -> tv_name.text as String + "\n秋月无边"
                else -> tv_name.text as String + "\n好诗！"
            }
            count = (count + 1) % 3
            true
        }
    }
}
